packages/compass-global-writes/src/plugin-title.tsx

```diff
@@ -1,5 +1,78 @@
+import { connect } from 'react-redux';
 import React from 'react';
+import { type RootState, ShardingStatuses } from './store/reducer';
+import {
+  Body,
+  css,
+  cx,
+  Icon,
+  palette,
+  spacing,
+  Tooltip,
+  useDarkMode,
+} from '@mongodb-js/compass-components';
 
-export function GlobalWritesTabTitle() {
-  return <div data-testid="global-writes-tab-title">Global Writes</div>;
-}
+const containerStyles = css({
+  display: 'flex',
+  gap: spacing[200],
+  alignItems: 'center',
+});
+
+const warningIconStyles = css({
+  display: 'flex',
+});
+
+const iconStylesLight = css({
+  color: palette.yellow.dark2,
+});
+
+const iconStylesDark = css({
+  color: palette.yellow.base,
+});
+
+const PluginTitle = ({ showWarning }: { showWarning: boolean }) => {
+  const darkMode = useDarkMode();
+  return (
+    <div data-testid="global-writes-tab-title" className={containerStyles}>
+      Global Writes{' '}
+      {showWarning && (
+        <Tooltip
+          data-testid="collection-stats-tooltip"
+          align="bottom"
+          justify="middle"
+          trigger={
+            // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+            <span
+              onClick={() => {
+                // LG does not bubble up the click event to the parent component,
+                // so we add noop onClick and let it bubble up.
+              }}
+            >
+              <Icon
+                glyph="ImportantWithCircle"
+                className={cx(
+                  warningIconStyles,
+                  iconStylesLight,
+                  darkMode && iconStylesDark
+                )}
+              />
+            </span>
+          }
+        >
+          <Body>
+            Collections in Atlas Global Clusters with Atlas-managed sharding
+            must be configured with a compound shard key made up of both a
+            &apos;location&apos; field and an identifier field that you provide.
+            Please configure sharding here.
+          </Body>
+        </Tooltip>
+      )}
+    </div>
+  );
+};
+
+export const GlobalWritesTabTitle = connect(
+  ({ isNamespaceSharded, status }: RootState) => ({
+    showWarning: !isNamespaceSharded && status !== ShardingStatuses.NOT_READY,
```

Comment:
why don't we check for `ShardingStatuses.UNSHARDED` directly here? and how come this doesn't show the warning by `ShardingStatuses.SUBMITTING_FOR_SHARDING` - does that fall under `isNamespaceSharded`? 🤔

Reply:
actually, it does make sense to use `ShardingStatuses.UNSHARDED` here which means for any state change we have to ensure we update the `isNamespaceSharded`. it should and it does show the warning its in  `SUBMITTING_FOR_SHARDING` state (the state when Submit For sharding button is disabled). Once the server accepts the request, we transition to `SHARDING` state, where this warning is not shown.

Reply:
~actually, it does make sense to use `ShardingStatuses.UNSHARDED` here which means for any state change we have to ensure we update the `isNamespaceSharded`.~ it should and it does show the warning its in  `SUBMITTING_FOR_SHARDING` state (the state when Submit For sharding button is disabled). Once the server accepts the request, we transition to `SHARDING` state, where this warning is not shown.